### PR TITLE
Responses to reviewer comments

### PR DESCRIPTION
**Pull request by kcranston**
First pass at reviewer comments. See also issue #170 and #169

### CODE REVIEW

File: good-enough-practices-for-scientific-computing.tex [hunk 1]
```diff
@@ -139,7 +139,7 @@
 \section*{Abstract}
 
 We present a set of computing tools and techniques that every
-researcher can and should adopt.  These recommendations synthesize
+researcher can adopt and should consider adopting.  These recommendations synthesize
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex]: I find the phrasing awkward - maybe just "can adopt." and drop the "should".

File: good-enough-practices-for-scientific-computing.tex [hunk 2]
```diff
@@ -150,13 +150,12 @@ \section*{Author Summary}
 
 Computers are now essential in all branches of science, but most
 researchers are never taught the equivalent of basic lab skills for
-research computing.  As a result, they take days or weeks to do things
-that could be done in minutes or hours, are often unable to reproduce
-their own work (much less the work of others), and have no idea how
-reliable their computational results are.
+research computing. As a result, their analyses can take much longer
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:153]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 3]
```diff
@@ -150,13 +150,12 @@ \section*{Author Summary}
...
+than necessary, and they may not be able to reproduce their work or
+make it reproducible by others. 
 
 This paper presents a set of good computing practices that every
-researcher can adopt regardless of their current level of technical
+researcher can adopt regardless of their current level of computational
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:158]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 4]
```diff
@@ -181,7 +180,7 @@ \section*{Introduction}\label{sec:introduction}
 practices\footnote{Note that English lacks a good word for this:
   ``mediocre'', ``adequate'', and ``sufficient'' aren't exactly
   right.} for scientific computing, i.e., a minimum set of tools and
-techniques that we believe every researcher can and should adopt. It
+techniques that we believe every researcher can adopt and should consider adopting. It
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex]: as above, just "can adopt" and drop the "should".

File: good-enough-practices-for-scientific-computing.tex [hunk 5]
```diff
@@ -190,10 +189,8 @@ \section*{Introduction}\label{sec:introduction}
 
 Our intended audience is researchers who are working alone or with a
 handful of collaborators on projects lasting a few days to a few
-months, and who are ready to move beyond emailing themselves a spreadsheet
-named \texttt{results-updated-3-revised.xlsx} at the end of the workday. A
-practice is included in our list if large numbers of researchers use
-it, and large numbers of people are \emph{still} using it months
+months. A practice is included in our list if large numbers of researchers 
+use it, and large numbers of people are \emph{still} using it months
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:193]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 6]
```diff
@@ -269,7 +266,11 @@ \subsection*{Acknowledgments}
   Some data will be impractical to manage in this way. For example, you should avoid
   making local copies of large, stable repositories.
   In that case, record the exact procedure used to obtain the raw data, as well
-  as any other pertinent info, such as an official version number or the date of download.
+  as any other pertinent information, such as an official version number or the date of download. 
+  
+  Ensure that raw data is backed up in more than one location. For large data sets,
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:271]: Make "Ensure raw data backed up" a `\practice`?

File: good-enough-practices-for-scientific-computing.tex [hunk 7]
```diff
@@ -292,7 +293,7 @@ \subsection*{Acknowledgments}
   \texttt{personal\_name} and \texttt{family\_name}, recode the
   treatment variable from \texttt{1} vs.  \texttt{2} to
   \texttt{untreated} vs. \texttt{treated}, and replace artificial
-  codes for missing data, such as ``-99'', with \texttt{NA}s,
+  codes for missing data, such as ``-99'', with \texttt{NA},
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:296]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 8]
```diff
@@ -328,8 +329,9 @@ \subsection*{Acknowledgments}
   The best way to do this is to write scripts for \emph{every} stage of data processing.
   This might feel frustratingly slow, but you will get faster with practice. The
   immediate payoff will be the ease with which you can re-do data preparation when new
-  data arrives. You can also re-use data preparation logic in the future for related
-  projects.
+  data arrives. You can also re-use data preparation steps in the future for related
+  projects. For very large data sets, data preparation may also include scripts to obtain
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:333]: "...data preparation may include writing and saving scripts to obtain the data (or subsets of the data) from remote storage."

File: good-enough-practices-for-scientific-computing.tex [hunk 9]
```diff
@@ -380,7 +382,9 @@ \subsection*{Acknowledgments}
 files makes it easy to re-run \emph{parts} of a data analysis pipeline,
 which in turn makes it less onerous to revisit and improve specific data
 processing tasks. Breaking a lengthy workflow into pieces makes it
-easier to understand, share, describe, and modify.
+easier to understand, share, describe, and modify. This is particularly 
+true when working with large data sets, where storage and transfer 
+of the entire data set is not trivial or inexpensive.
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:387]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 10]
```diff
@@ -396,7 +400,10 @@ \subsection*{Acknowledgments}
 The core realization in these practices is that \emph{readable},
 \emph{reusable}, and \emph{testable} are all side effects of writing
 \emph{modular} code, i.e., of building programs out of short,
-single-purpose functions with clearly-defined inputs and outputs.
+single-purpose functions with clearly-defined inputs and outputs. Much 
+has been written on this topic (refs!), and this section focuses on practices
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:404]: I'll find refs.

File: good-enough-practices-for-scientific-computing.tex [hunk 11]
```diff
@@ -396,7 +400,10 @@ \subsection*{Acknowledgments}
 The core realization in these practices is that \emph{readable},
 \emph{reusable}, and \emph{testable} are all side effects of writing
 \emph{modular} code, i.e., of building programs out of short,
-single-purpose functions with clearly-defined inputs and outputs.
+single-purpose functions with clearly-defined inputs and outputs. Much 
+has been written on this topic (refs!), and this section focuses on practices 
+that best balance ease of employment with benefit for you and
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:405]: "benefit for you and collaborations" sounds odd.

File: good-enough-practices-for-scientific-computing.tex [hunk 12]
```diff
@@ -536,7 +544,7 @@ \subsection*{Acknowledgments}
   to.
 
 \item
-  \practice{3b}{Create a shared public ``to-do'' list}.  This can be a
+  \practice{3b}{Create a shared ``to-do'' list}.  This can be a
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:547]: Reluctant :+1: (I understand not everyone believes in all-open, all the time)

File: good-enough-practices-for-scientific-computing.tex [hunk 13]
```diff
@@ -545,7 +553,16 @@ \subsection*{Acknowledgments}
...
 
 \item
-  \practice{3c}{Make the license explicit.}  Have a \texttt{LICENSE} file
+  \practice{3c}{Decide on communication strategies.} Make explicit
+  decisions about (and publicize where appropriate) how members 
+  of the project will communicate with
+  each other and with externals users / collaborators. This includes
+  the location and technology for email lists, chat channels, voice / 
+  video conferencing, documentation, and meeting notes, as well as 
+  which of these channels will be public or private.
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:562]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 14]
```diff
@@ -645,7 +662,10 @@ \subsection*{Acknowledgments}
   tables, and final publication-ready figures or tables should be
   separated clearly by file naming conventions or placed into
   different subdirectories; those belonging to different papers or
-  other publications should be grouped together.
+  other publications should be grouped together. Similarly, the
+  \texttt{data} directory might require subdirectories to organize 
+  raw data based on time, method of collection, or other 
+  metadata.
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:668]: Should we talk about how to name subdirectories?

File: good-enough-practices-for-scientific-computing.tex [hunk 15]
```diff
@@ -812,8 +832,8 @@ \subsection*{Acknowledgments}
 
 \item
   \practice{5e}{Store each project in a folder that is mirrored off the
-    researcher's working machine} using a system such as Dropbox
-    or a remote version control repository such as GitHub.
+    researcher's working machine} using a system such as \withurl{Dropbox}{http://dropbox.com}
+    or a remote version control repository such as \withurl{GitHub}{http://github.com}.
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:836]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 16]
```diff
@@ -920,13 +940,7 @@ \subsection*{Version Control Systems}
...
 \withurl{GitLab}{https://about.gitlab.com} is a well-regarded
 alternative that some prefer, because the GitLab platform itself is
-free and open source.
-
-For those who find Git's command-line syntax inconsistent
-and confusing, \withurl{Mercurial}{https://www.mercurial-scm.org/} is
-a good choice; \withurl{Bitbucket}{https://bitbucket.org/} provides
-free hosting for both Git and Mercurial repositories, but does not
-have nearly as many scientific users.
+free and open source.
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:943]: I'm OK with removing mention of Mercurial, but think it's unfair to remove mention of BitBucket.

File: good-enough-practices-for-scientific-computing.tex [hunk 17]
```diff
@@ -983,7 +997,8 @@ \subsection*{What Not to Put Under Version Control}
 A common practice in academic writing is for the lead author to send
 successive versions of a manuscript to coauthors to collect feedback,
 which is returned as changes to the document, comments on the
-document, plain text in email, or a mix of all three. This results in
+document, plain text in email, or a mix of all three. This allows co-authors
+to use familiar tools, but results in
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:1001]: :+1:

File: good-enough-practices-for-scientific-computing.tex [hunk 18]
```diff
@@ -1141,8 +1156,11 @@ \subsection*{Why Two Recommendations for Manuscripts?}
 
 In keeping with our goal of recommending ``good enough'' practices, we
 have therefore included online, collaborative editing in something like Google
-Docs. We still recommend \emph{against} traditional desktop tools like
-LibreOffice and Microsoft Word because they make collaboration more
+Docs. This solution meets the requirements that we've outlined above, although 
+you do lose some benefits of text-based formats (notably, being able to use 
+non-proprietary tools for editing, sharing, and viewing and being amenable to  
+version control). We still recommend \emph{against} traditional desktop tools like
+LibreOffice and Microsoft\textregistered Word because they make collaboration more
```
Comment (gvwilson) [inline good-enough-practices-for-scientific-computing.tex:1163]: :+1:

File: good-enough-practices-for-scientific-computing.tex (hunk 12 above)
Comment (kcranston) [reply good-enough-practices-for-scientific-computing.tex:547]: This was in response to the reviewer who thought this whole section only applied to open source projects. I think these practices also apply when working privately (i.e. if new collaborators are only people joining the lab).

**Comment by gvwilson** [comment]
:+1:

**Comment by jennybc** [comment]
Thanks @kcranston. 👍

File: good-enough-practices-for-scientific-computing.tex (hunk 11 above)
Comment (lexnederbragt) [reply good-enough-practices-for-scientific-computing.tex:405]: your collaborators?

File: good-enough-practices-for-scientific-computing.tex (hunk 14 above)
Comment (lexnederbragt) [reply good-enough-practices-for-scientific-computing.tex:668]: Not sure what you mean, can anything be said that is general enough?

File: good-enough-practices-for-scientific-computing.tex [hunk 19]
```diff
@@ -878,8 +898,8 @@ \subsection*{Manual Versioning}
     \\
     Copying everything like this may seem wasteful, since many files
     won't have changed, but consider: a terabyte hard drive costs
-    about \$50 retail, which means that 50 GByte costs less than a
-    latte. Provided large data files are kept out of the backed-up
+    about \$50 retail, which means that 50 GByte costs less than 
+    \$5. Provided large data files are kept out of the backed-up
```
Comment (lexnederbragt) [inline good-enough-practices-for-scientific-computing.tex:902]: 👍

File: good-enough-practices-for-scientific-computing.tex (hunk 16 above)
Comment (lexnederbragt) [reply good-enough-practices-for-scientific-computing.tex:943]: I agree. Can we keep BitBucket?

**Comment by tracykteal** [comment]
Since there are enough unresolved comments, it doesn't seem like this can be merged. So, I'm going to work in this branch to address the comments and the other ones others have mentioned in the emails. 

Let me know if there's an objection to me doing the updates in this branch.

**Comment by tracykteal** [comment]
Note that a comment thread on an individual line does go away when that section gets updated with new PRs.

**Comment by gvwilson** [comment]
+1

**Comment by tracykteal** [comment]
I addressed the issues here, but I also made quite a few other updates. So, I'm going to merge this PR, and then put in a new one.

**Pull request merged**